File: crates/conformance/golden/swift/Sample.swift
import Foundation

struct Point {
  let x: Int
  let y: Int
}

enum Shape {
  case circle(radius: Double)
  case rect(w: Double, h: Double)
}

func area(of shape: Shape) -> Double {
  switch shape {
    case .circle(let radius):
  return 3.14159 * radius * radius
    case .rect(let w, let h):
  return w * h
  }
}

func main() {
  let points = [Point(x: 1, y: 2), Point(x: 3, y: 4)]
  let sum = points.map { p in
    p.x + p.y
  }.reduce(0) { acc, v in
    acc + v
  }
  print("sum=\(sum)")

  let banner = """
    hello
      world
    """
  print(banner)

  for i in 0..<3 {
    if i % 2 == 0 {
      print("even \(i)")
    } else {
      print("odd \(i)")
    }
  }
}

File: crates/conformance/corpus/zig/sample.zig
const std = @import("std");

const Shape = union(enum) {
    circle: f64,
    rect: struct { w: f64, h: f64 },

    fn area(self: Shape) f64 {
        return switch (self) {
            .circle => |r| 3.14159 * r * r,
            .rect => |dims| dims.w * dims.h,
        };
    }
};

pub fn main() void {
    const shapes = [_]Shape{
        .{ .circle = 2.0 },
        .{ .rect = .{ .w = 3.0, .h = 4.0 } },
    };

    var total: f64 = 0;
    for (shapes) |s| {
        total += s.area();
    }

    var i: usize = 0;
    while (i < 3) : (i += 1) {
        if (i % 2 == 0) {
            std.debug.print("even {d}\n", .{i});
        } else {
            std.debug.print("odd {d}\n", .{i});
        }
    }

    std.debug.print("total={d}\n", .{total});
}
